src/main/java/org/example/intro/repository/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,36 @@
+package org.example.intro.repository;
+
+import lombok.RequiredArgsConstructor;
+import org.example.intro.dto.BookSearchParametersDto;
+import org.example.intro.model.Book;
+import org.example.intro.repository.book.SpecificationBuilder;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookBookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParameters) {
+        System.out.println("searchParameters: " + searchParameters);
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            spec = spec.and(bookBookSpecificationProviderManager
+                    .getSpecificationProvider("title")
```

Comment:
Create constants for title, author,  isbn

Reply:
++